src/common/io.ts

```diff
@@ -616,6 +628,23 @@ export class ARIOReadable implements AoARIORead {
   }: AoGetCostDetailsParams): Promise<CostDetailsResult> {
     const replacedBuyRecordWithBuyName =
       intent === 'Buy-Record' ? 'Buy-Name' : intent;
+
+    if (fundFrom === 'turbo') {
+      const { mARIO, winc } = await this.fundFromTurbo.getArNSPurchasePrice({
+        intent: replacedBuyRecordWithBuyName,
+        name,
+        increaseQty: quantity,
+        type,
+        years,
+      });
+
+      return {
+        tokenCost: +mARIO,
+        wincQty: winc,
+        discounts: [],
+      };
```

Comment:
should this have an expiry timestamp?

Reply:
no this is just grabbing the current price from payment service. up next on payment service I'll be working on the arnsPurchaseQuote endpoint that will have a purchaseQuoteExpiration on the entity. that will be fulfilled by using a credit card to stripe